src/app/utils/file-name-helpers.ts

```diff
@@ -0,0 +1,21 @@
+export function shortenFileName(
+  fileName: string,
+  maxLength: number = 20,
+): string {
+  if (!fileName || fileName.length <= maxLength) {
+    return fileName;
+  }
+
+  const extension = fileName.includes('.')
+    ? fileName.slice(fileName.lastIndexOf('.'))
+    : '';
+  const nameWithoutExtension = fileName.slice(0, fileName.lastIndexOf('.'));
+
+  const halfLength = Math.floor((maxLength - 3 - extension.length) / 2);
+  const start = nameWithoutExtension.slice(0, halfLength);
```

Comment:
let's combine the file-name-helpers and file-upload-helpers into `file-helpers.ts`